src/qibo/noise_model.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+from qibo import gates, models
+from qibo.noise import DepolarizingError, NoiseModel, ThermalRelaxationError
+
+
+def noise_model(Circuit, params):
+    """Creates a noisy circuit from the circuit given as argument.
+
+    The function applies a :class:`qibo.gates.ThermalRelaxationChannel` after each step of the circuit
+    and, after each gate, a :class:`qibo.gates.DepolarizingChannel`, whose parameter depends on whether the
+    gate applies on one or two qubits. In the end on the samples are applied bitflips errors using
+    the :class:`qibo.gates.PauliNoiseChannel`.
+
+
+    Args:
+        Circuit (qibo.models.Circuit): Circuit on which noise will be applied. Since in the end are
+        applied bitflips, measurement gates are required.
+        Params (tuple): object which contains the parameters of the channels organized as follow
+        ``((((t1,t2),...,(t1,t2)),(time_1,time_2),excited_population),(lam_1,lam_2),(px_1,...,px_m))``.
+        The first element is a tuple which contains the parameters of the :class:`qibo.gates.ThermalRelaxationChannel`. The second element
+        is a tuple which contains the parameter of the :class:`qibo.gates.DepolarizingChannel` for 1-qubit gate and 2-qubit gate, respectively.
+        The third element is a dictionary "error map" for the bit flips.
+        nshots (int): number of final samples.
+
+    Returns:
+        Circuit (qibo.models.Circuit) padded with noise channels.
+
+
+    """
+    Noisy_Circuit = Circuit.__class__(**Circuit.init_kwargs)
+
+    time_steps = max(Circuit.queue.moment_index)
+    time = time_steps * [
+        0,
+    ]
+    current_time = Circuit.nqubits * [
+        0,
+    ]
+    t1 = params[0][1][0]
```

Comment:
I find this mechanism quite complex to understand. You might prefer to use a dictionary or multiple parameters to avoid 2 or 4 parameters.